File: 스택/스택.py
import sys
N = int(input())
stack = []

for i in range(N):
    prompt = sys.stdin.readline().rstrip()
    try:
        prompt_str, prompt_int = prompt.split()
    except:
        prompt_str = prompt
    if (prompt_str == "push"):
        stack.append(int(prompt_int))
    elif (prompt_str == "pop"):
        if (stack):
            print(stack[-1])
            stack.pop()
        else:
            print(-1)
    elif (prompt_str == "size"):
        print(len(stack))
    elif (prompt_str == "empty"):
        if (stack):
            print(0)
        else:
            print(1)
    elif (prompt_str == "top"):
        if (stack):
            print(stack[-1])
        else:
            print(-1)

File: 알고리즘 스터디 문제들/Part 2 - 자료구조/문제/스택.cpp
#include <iostream>
#include <string>
using namespace std;

struct Stack
{
    int arr[10000];
    int top = -1;
};

void push(Stack &S, int X)
{
    S.arr[++S.top] = X;
}

int pop(Stack &S)
{
    if (S.top == -1)
        return -1;
    else
        return S.arr[S.top--];
}

bool empty(Stack &S)
{
    return S.top == -1;
}

int top(Stack &S)
{
    if (S.top == -1)
        return -1;
    else
        return S.arr[S.top];
}

int size(Stack &S)
{
    return S.top + 1;
}

// Class 구현도 가능하지만 일단은 struct만으로 구현해보자.
int main()
{
    size_t N;
    cin >> N;

    Stack S;
    string input;
    for (size_t i = 0; i < N; i++)
    {
        cin >> input;

        if (input == "push")
        {
            int tmp;
            cin >> tmp;
            push(S, tmp);
        }
        else if (input == "pop")
            cout << pop(S) << '\n';
        else if (input == "size")
            cout << size(S) << '\n';
        else if (input == "empty")
            cout << empty(S) << '\n';
        else if (input == "top")
            cout << top(S) << '\n';
    }
}